src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java

```diff
@@ -285,17 +296,37 @@ private void runAnalyzerEngine() throws AbortException {
     } catch (AbortException e) { // probably caught one of the thrown exceptions, let it pass through
       throw e;
     } catch (Exception e) { // caught unknown exception, log it and wrap its
-      console.logError("Failed to add image(s) to anchore-engine due to an unexpected error", e);
+      console.logError("Failed to add image(s) to anchore-enterprise due to an unexpected error", e);
       throw new AbortException(
-          "Failed to add image(s) to anchore-engine due to an unexpected error. Please refer to above logs for more information");
+          "Failed to add image(s) to anchore-enterprise due to an unexpected error. Please refer to above logs for more information");
     }
   }
 
+  private void writeResponseToFile(Integer counter, FilePath jenkinsOutputDirFP, String responseBody) throws AbortException {
+    // Write api response to a file as it is
+    String jenkinsAEResponseFileName = AE_EVAL_PREFIX + (counter) + JSON_FILE_EXTENSION;
+    FilePath jenkinsAEResponseFP = new FilePath(jenkinsOutputDirFP, jenkinsAEResponseFileName);
+
+    try {
+      console.logDebug("Writing anchore-enterprise policy evaluation response to " + jenkinsAEResponseFP.getRemote());
+      try (BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(jenkinsAEResponseFP.write(), StandardCharsets.UTF_8))) {
+        bw.write(responseBody);
+      }
+    } catch (IOException | InterruptedException e) {
+      console.logWarn("Failed to write anchore-enterprise policy evaluation response to " + jenkinsAEResponseFP.getRemote(), e);
+      throw new AbortException("Failed to write anchore-enterprise policy evaluation response to " + jenkinsAEResponseFP.getRemote());
+    }
+
+  }
+
   public GATE_ACTION runGates() throws AbortException {
-    return runGatesEngine();
+    if (config.getEngineApiVersion() == API_VERSION.v1) {
+      return runGatesEngineV1();
+    }
```

Comment:
Can we add some logs stating the API version in use if this code path is executed, just so it's easier to debug if a future version breaks this code path. There may be somewhere better to add the logs that here but your call